program-tests/pre-authorized-debit-v1/debit/oneTime.shared.ts

```diff
@@ -423,6 +423,154 @@ export function testOneTimeDebit(
       });
     });
 
+    context("negative timestamps", () => {
+      it("fails if expiry is negative", async () => {
+        await program.methods
+          .closePreAuthorization()
+          .accounts({
+            receiver: userKeypair.publicKey,
+            authority: userKeypair.publicKey,
+            tokenAccount: tokenAccountPubkey,
+            preAuthorization: preAuthorizationPubkey,
+          })
+          .signers([userKeypair])
+          .rpc();
+
+        const activationUnixTimestamp =
+          Math.floor(new Date().getTime() / 1e3) - 24 * 60 * 60; // -1 day before now
+
+        const expirationUnixTimestamp = -(
+          activationUnixTimestamp +
+          10 * 24 * 60 * 60 -
+          1
+        ); // negative(+10 days from now) (so thousands of year before epoch 0)
+
+        await setupOneTimePreAuthorization(
+          activationUnixTimestamp,
+          expirationUnixTimestamp,
+        );
+
+        await expect(
+          program.methods
+            .debit({ amount: new anchor.BN(50e6) })
+            .accounts({
+              debitAuthority: debitAuthorityKeypair.publicKey,
+              mint: mintPubkey,
+              tokenAccount: tokenAccountPubkey,
+              destinationTokenAccount: destinationTokenAccountPubkey,
+              smartDelegate: smartDelegatePubkey,
+              preAuthorization: preAuthorizationPubkey,
+              tokenProgram: tokenProgramId,
+            })
+            .signers([debitAuthorityKeypair])
+            .rpc(),
+        ).to.eventually.be.rejectedWith(
+          /Error Code: PreAuthorizationNotActive. Error Number: 6000/,
+        );
+      });
+      it("fails if activation is negative but quantitatively larger than past expiry", async () => {
```

Comment:
Nit: this test title is a little confusing.

Reply:
fixed